File: src/main/java/game/Tile.java
package game;

public class Tile {
    private final int x;
    private final int y;
    private Symbol symbol = Symbol.EMPTY;

    public Tile(int x, int y) {
        this.x = x;
        this.y = y;
    }

    boolean isAt(int x, int y) {
        return this.x == x && this.y == y;
    }

    boolean isMarked() {
        return symbol.isNotEmpty();
    }

    boolean hasSameSymbolAs(Tile other) {
        return symbol == other.symbol;
    }

    void mark(Symbol symbol) {
        this.symbol = symbol;
    }

    public Symbol symbol() {
        return symbol;
    }
}

File: src/main/java/game/Tile.kt
package game

class Tile(val x: Int, private val y: Int) {

    var symbol = Symbol.BLANK
        private set

    val isMarked get() = symbol.isNotBlank

    fun isAt(x: Int, y: Int) = this.x == x && this.y == y

    fun hasSameSymbolAs(other: Tile) = symbol == other.symbol

    fun mark(symbol: Symbol) { this.symbol = symbol }
}
